src/generate/register.rs

```diff
@@ -263,13 +315,13 @@ pub fn render_register_mod(
         }
     }
 
-    let open = Punct::new('{', Spacing::Alone);
-    let close = Punct::new('}', Spacing::Alone);
-
     if can_read && !r_impl_items.is_empty() {
         mod_items.extend(quote! {
             impl R #open #r_impl_items #close
         });
+        mod_items.extend(quote!{
+            #r_debug_impl
+        });
```

Comment:
This should not be added by default. Need new option:
```suggestion
        if config.register_core_debug {
            mod_items.extend(quote!{
                #r_debug_impl
            });
        }
```

Reply:
@burrbull  - This adds `impl Debug` for read registers gated by `#[cfg(feature="impl-register-debug")]` in the resulting pac so that users of the pac can choose to have debug or no debut without having to generate the pac themselves.

Reply:
This significantly grows `pac` codebase with dead code for most of users. So no, this should not be generated by default.

Also we should have possibility to change core `Debug` with `defmt::Format` which looks more usable in this case.

Reply:
I've added options to enable Debug and optionally specify a gating feature.

As for `defmt::Format` I've not worked with `defmt` and could use some hints on how you think the data should be formatted?  For Debug I used the `debug_struct` in `core::fmt::Formatter`.